control-plane/api-gateway/gatekeeper/service.go

```diff
@@ -65,14 +65,22 @@ func (g *Gatekeeper) deleteService(ctx context.Context, gwName types.NamespacedN
 }
 
 func (g *Gatekeeper) service(gateway gwv1beta1.Gateway, gcc v1alpha1.GatewayClassConfig) *corev1.Service {
+	seenPorts := map[gwv1beta1.PortNumber]struct{}{}
 	ports := []corev1.ServicePort{}
 	for _, listener := range gateway.Spec.Listeners {
+		if _, seen := seenPorts[listener.Port]; seen {
+			// We've already added this listener's port to the Service
```

Comment:
Should this be throwing some kind of error? It seems to me like its a configuration issue if multiple listeners are defined on the same port.

Reply:
The Gateway API spec makes a specific allowance for multiple listeners to use the same port ([source](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.Listener)):

> Port is the network port. Multiple listeners may use the same port, subject to the Listener compatibility rules.